lib/internal/modules/cjs/loader.js

```diff
@@ -573,28 +574,33 @@ Module._resolveFilename = function(request, parent, isMain, options) {
 
   var paths;
 
-  if (typeof options === 'object' && options !== null &&
-      Array.isArray(options.paths)) {
-    const isRelative = request.startsWith('./') || request.startsWith('../') ||
-        (isWindows && request.startsWith('.\\') || request.startsWith('..\\'));
+  if (typeof options === 'object' && options !== null) {
+    if (Array.isArray(options.paths)) {
```

Comment:
I suggest to turn this check around. That way it's less indentation and thus less churn.

Reply:
I'm not sure that would end up being correct. I think the `paths` handling logic needs to be inside an `Array.ifArray()` check. Otherwise, if an empty options object is passed in, or if additional options are supported in the future, then this code would execute when it shouldn't.

Reply:
I would write it as:

```js
if (options.paths !== undefined && !Array.isArray(options.paths)) {
  throw new ...
}
const isRelative = request.startsWith(...
```

Reply:
The code starting with `const isRelative = ...` should not be executed if `options.paths` is `undefined`, even though that is a valid value for `options.paths`.

Reply:
Oh, off course! I overlooked that. Ignore my comment :-)